src/stdlib_math_diff.fypp

```diff
@@ -38,16 +37,24 @@ contains
             return
         end if
 
-        allocate(work(size_x + size_prepend + size_append))
+        #! Use a quick exit for the common case, to avoid memory allocation.
```

Comment:
Does the `#` need to be removed?

Reply:
Actually it is legal (thanks to fypp). But I suggest removing anyway.

Reply:
`#!` is [the comment header of `fypp`](https://fypp.readthedocs.io/en/stable/fypp.html#comment-directive), which can be replaced by `!`.

Reply:
`#!` is [the comment header of `fypp`](https://fypp.readthedocs.io/en/stable/fypp.html#comment-directive), which can be replaced by `!`.